src/pages/stops/stops.ts

```diff
@@ -15,24 +19,36 @@ import { StopDetailsPage } from '../stop-details/stop-details';
   templateUrl: 'stops.html',
 })
 export class StopsPage {
-  items: Array<{title: string}>;
+  /*
+    items: Array<{ title: string }>;
+    constructor(public navCtrl: NavController, public modalCtrl: ModalController, public navParams: NavParams) {
+      this.items = [];
+      let item1 = { title: 'Weber Arch' };
+      let item2 = { title: 'Tech Institute' };
+      this.items.push(item1);
+      this.items.push(item2);
+    }
+  */
 
-  constructor(public navCtrl: NavController, public modalCtrl: ModalController, public navParams: NavParams) {
-    this.items = [];
-    let item1 = {title: 'Weber Arch'};
-    let item2 = {title: 'Tech Institute'};
-    this.items.push(item1);
-    this.items.push(item2);
+  items: Observable<any[]>;
+  // busStopsRef: AngularFireList<any>;
+  constructor(public navCtrl: NavController, public modalCtrl: ModalController, public navParams: NavParams, public firebaseProvider: FirebaseProvider) {
+    // this.busStopsRef = this.firebaseProvider.getBusStops();
+    // this.busStops = this.busStopsRef.valueChanges();
+    this.items = this.firebaseProvider.getBusStops().valueChanges();
   }
 
   ionViewDidLoad() {
     console.log('ionViewDidLoad StopsPage');
+    // var time = new Date().getDay();
+    // console.log(time);
   }
 
   itemTapped(event, item) {
     let stopDetailsModal = this.modalCtrl.create(StopDetailsPage,
-    {stopName: item.title},
-    {cssClass:'stopDetailsModal'});
+      { stopName: item.name, stopLine: item.line, stopID: item.id },
```

Comment:
Do we need the stop line to be passed in? Just the stop should be sufficient right?

Reply:
agreed, do not need to pass in the stopLine and stopName. Just the stop id will be right